### ISSUE TITLE
Project broken in modern django

### ISSUE THREAD

**Issue by fake-name** [peterkmurphy/glyphviewer#1]
Apparently this is broken in django > 1.9. In particular, a bunch of stuff in `urls.py` is no longer valid.

 - `django.conf.urls.defaults` no longer exists.  
 - `django.conf.urls.patterns()` no longer exists too.
 - URLPatterns apparently are now supposed to take a callable?

It'd be nice if there was a bootstrap script too. I'm interested in just checking a few things with `fonttools`, and I've never worked with django at all, so just getting the project up and running is a confusing and complex process.

Simplest solution (though not elegant) would be to hard-code the django max version, but in any event, `"Django >= 1.0"` in `setup.py` is certainly not valid.

**Comment by peterkmurphy** [comment]
@fake-name: the issues you have affect some other Django projects I have, so I will be looking at those as well. I am aware that the dependencies are a little screwed, so I will try to clean them up.

Could you however explain what you mean or expect by a "bootstrap script"? Glyphviewer depends on Python; are you looking for a project without a Python dependency?

Best regards,
Peter

**Comment by fake-name** [comment]
No, I mean, how do I even run the thing?

I haven't worked with django before. Apparently, I have to set up django environment or something, and somehow install glyphviewer into it? 

Basically, a script where you run it, and it runs the web-application without having anything installed beforehand. 

Basically, when I've done this sort of thing for my projects, it winds up being a script that:
 1. Checks for a venv in the current directory.
     2. If the venv doesn't exist, creates it.
     3. Install the dependencies into the venv.
     4. Do any needed setup to make the project launchable.
 2. Now that we have a run environment, start the project.

(Example [pt 1](https://github.com/fake-name/AutoTriever/blob/master/run.sh) and [pt 2](https://github.com/fake-name/AutoTriever/blob/master/configure.sh))

Basically, a script that will run the project, with no prerequisites (other then python & pip).

I never did get this project to *do* anything, because apparently you need to know a fair bit about how to run a django project to make it go, and I didn't have the time.

Of course, doing follow up research now, I think [`django-admin startproject mysite`](https://docs.djangoproject.com/en/1.9/ref/applications/) creates the skeleton (why didn't I find that previously!), but the point is, right now, the project isn't just runnable without futzing about with the django documentation. 

Since poking around in font-files is of interest to people outside of web-app development, I think having a simple quickstart that gets the project running without any library-specific knowledge would be tremendously valuable.

**Comment by peterkmurphy** [comment]
@fake-name: I think I get you. I intended the application to be tacked onto an existing Django site (which needs some knowledge to make it run). At the moment, it's tweaked to work with Mezzanine, a CMS based on Django. But if you could run a script that generates a simple, standalone website that sets up glyphviewer, that would be useful as well. I think that is definitely something to aim for.

**Comment by fake-name** [comment]
Yeah, basically, I found the project because I was doing some horrible hackery on truetype fonts, and there's surprisingly few easily available tools for that. My interest was purely for using it as a simple font-viewer. (There are wacko's on the internet "protecting" HTML content by using custom fonts with randomized code-point orders, and I was trying to build a system to reverse that).

I would generally assume that a "web app" is something capable of stand-alone execution. It may be pedantic, but I'd classify this project as a web-application *module* at the moment, since it's not stand-alone.

<sub>(I could go off on a rant about how I think web-application frameworks and projects in general that have their own launcher (such as `django-admin`) is a horrible, horrible anti-pattern since it basically precludes using the library as a module in a larger project, but that's off topic for the issue, and is mostly just me being angry on the internet).</sub>

I eventually wound up just putzing about in the internals of `fontTools.ttLib` and doing a bunch of hand-matching of the vector primitives, but that's very very non-ideal.
